core/src/impl/Kokkos_ViewHooks.hpp

```diff
@@ -0,0 +1,475 @@
+    }
+    if (ptr != nullptr) delete ptr;
+  }
+
+  template <typename F>
+  static void clear(const std::string &name, ViewHookCaller<F, void> *ptr) {
+    if (s_map_callers.find(name) != s_map_callers.end()) {
+      s_map_callers.erase(name);
+    }
+    if (ptr != nullptr) delete ptr;
+  }
+
+  template <typename F, typename ConstF>
+  static void clear(const std::string &name,
+                    ViewHookCopyCaller<F, ConstF> *ptr) {
+    if (s_map_callers.find(name) != s_map_callers.end()) {
+      s_map_callers.erase(name);
+    }
+    if (ptr != nullptr) delete ptr;
+  }
+
+  template <typename F>
+  static void clear(const std::string &name, ViewHookCopyCaller<F, void> *ptr) {
+    if (s_map_callers.find(name) != s_map_callers.end()) {
+      s_map_callers.erase(name);
+    }
+    if (ptr != nullptr) delete ptr;
+  }
+  static void clear() { s_map_callers.clear(); }
+
+  static bool is_set() noexcept { return !(s_map_callers.empty()); }
+
+  template <class DataType, class... Properties>
+  static void call(const View<DataType, Properties...> &view) {
+    using view_holder_type =
+        ViewHolderCopy<const View<DataType, Properties...> >;
+    view_holder_type holder(view);
+
+    for (const auto &caller : s_map_callers) {
+      if (!caller.second->apply_to_copy_construct()) {
+        caller.second->do_call(holder);
+      }
+    }
+  }
+
+  template <class DataType, class... Properties>
+  static typename std::enable_if<
+      (!std::is_same<
+           typename Kokkos::ViewTraits<DataType, Properties...>::value_type,
+           typename Kokkos::ViewTraits<
+               DataType, Properties...>::const_value_type>::value &&
+       !std::is_same<
+           typename Kokkos::ViewTraits<DataType, Properties...>::memory_space,
+           Kokkos::AnonymousSpace>::value),
+      void>::type
+  call(View<DataType, Properties...> &dst,
+       const View<DataType, Properties...> &src) {
+    using non_const_view_holder = ViewHolderRef<View<DataType, Properties...> >;
+    using const_view_holder =
+        ViewHolderCopy<const View<DataType, Properties...> >;
+    const_view_holder src_holder(src);
+    non_const_view_holder dst_holder(dst);
+
+    for (const auto &caller : s_map_callers) {
+      if (caller.second->apply_to_copy_construct()) {
+        caller.second->do_call(dst_holder, src_holder);
+      } else {
+        caller.second->do_call(src_holder);
+      }
+    }
+  }
+
+  // we have to treat anonymous space different...both have to be const...
+  template <class DataType, class... Properties>
+  static typename std::enable_if<
+      (std::is_same<
+           typename Kokkos::ViewTraits<DataType, Properties...>::value_type,
+           typename Kokkos::ViewTraits<
+               DataType, Properties...>::const_value_type>::value ||
+       std::is_same<
+           typename Kokkos::ViewTraits<DataType, Properties...>::memory_space,
+           Kokkos::AnonymousSpace>::value),
+      void>::type
```

Comment:
Please don't copy-paste this condition. If it's different from the one above (I can't tell at a glance), indicate the difference in a comment.